### PR TITLE
KOA-4768 migrate to SceneDelegate

### PR DESCRIPTION
**Pull request by jookes**
Migrate the AppDelegate from ObjectiveC to Swift and split into the newer SceneDelegate

+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] `UNRELEASED.md`
+ [ ] `README.md`
+ [ ] Tests
+ [ ] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [ ] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)
+ [ ] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Example/Backpack/SceneDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+import UIKit
+import Backpack
+
+class SceneDelegate: UIResponder, UIWindowSceneDelegate {
+
+    var window: UIWindow?
+    var isUITestingEnabled = false
+
+    // swiftlint:disable line_length
```
Comment (zgeor) [inline Example/Backpack/SceneDelegate.swift:28]: Any reason to disable the lint rule?

Comment (jookes) [reply Example/Backpack/SceneDelegate.swift:28]: When I split it into multiple lines, it will complain about indentation :/.

Comment (zgeor) [reply Example/Backpack/SceneDelegate.swift:28]: Can the indentation be fixed? Or are we using an incompatible ruleset?

Comment (jookes) [reply Example/Backpack/SceneDelegate.swift:28]: With the current ruleset we either:
- hit `line_length`
- hit `indentation_width`
But when we solve indentation_width, we will hit `vertical_parameter_alignment`

@frugoman What's your opinion on this?
We could disable the `line_length` or `indentation_width`, both would be fair game. But we can't get around both of these at the same time :( 

Relevant discussion:
https://github.com/realm/SwiftLint/issues/3046

Comment (frugoman) [reply Example/Backpack/SceneDelegate.swift:28]: In these cases I do commonly prefer this notation
```swift
func scene(
   _ scene: UIScene,
   willConnectTo session: UIScene,
   ....
) {
   //code here
}
```
not sure if there's a rule against it tho

**Review by frugoman — APPROVED** [review]
Looking good!

**Pull request merged**